Compile problem with new test tst_h_par_compress.c

A report from George at NOAA of a compile issue:

```
mpiicc -DHAVE_CONFIG_H -I. -I..  -I../include  -I/home/gvandenb/w/tmp/include   -MT tst_h_par_compress.o -MD -MP -MF $depbase.Tpo -c -o tst_h_par_compress.o tst_h_par_compress.c &&\
mv -f $depbase.Tpo $depbase.Po
tst_h_par_compress.c(46): error: expected an expression
      for (int cf = 0; cf < NUM_COMPRESS_FILTERS; cf++)
           ^

tst_h_par_compress.c(46): error: identifier "cf" is undefined
      for (int cf = 0; cf < NUM_COMPRESS_FILTERS; cf++)
                       ^

compilation aborted for tst_h_par_compress.c (code 2)
make[2]: *** [tst_h_par_compress.o] Error 2
make[2]: Leaving directory `/work/noaa/noaatest/gwv/netp/netcdf-c/h5_test'
make[1]: *** [check-am] Error 2
make[1]: Leaving directory `/work/noaa/noaatest/gwv/netp/netcdf-c/h5_test'
make: *** [check-recursive] Error 1
Orion-login-2[133] gvandenb$ find . -type f -print  | grep  tst_h_par_compress.c
./h5_test/tst_h_par_compress.c

tst_h_par_compress.c  is not in the 4.7.4 distribution..    This compiled on WCOSS2  It does not compile on orion or the wcoss dell.  IT DOES COMPILE ON WCOSS2

```

--- a/h5_test/tst_h_par_compress.c
+++ b/h5_test/tst_h_par_compress.c
@@ -35,6 +35,7 @@
 int
 main(int argc, char **argv)
 {
+    int cf;
     int p, my_rank;
 
     MPI_Init(&argc, &argv);
@@ -43,7 +44,7 @@ main(int argc, char **argv)
 
     /* For builds with HDF5 prior to 1.10.3, just return success. */
 #ifdef HDF5_SUPPORTS_PAR_FILTERS
-    for (int cf = 0; cf < NUM_COMPRESS_FILTERS; cf++)
+    for (cf = 0; cf < NUM_COMPRESS_FILTERS; cf++)
     {
 	if (!my_rank)
 	    printf("*** Testing parallel I/O with %s compression...", cf ? "szip" : "zlib");

